third_party/2and3/google/protobuf/internal/extension_dict.pyi

```diff
@@ -0,0 +1,22 @@
+from typing import Any, Generic, Iterator, TypeVar
+
+from google.protobuf.descriptor import FieldDescriptor
+from google.protobuf.message import Message
+
+_ContainerMessageT = TypeVar("_ContainerMessageT", bound=Message)
+_ExtenderMessageT = TypeVar("_ExtenderMessageT", bound=Message)
+
+class _ExtensionFieldDescriptor(FieldDescriptor, Generic[_ContainerMessageT, _ExtenderMessageT]): ...
+
+class _ExtensionDict(Generic[_ContainerMessageT]):
+    def __init__(self, extended_message: _ContainerMessageT) -> None: ...
+    def __getitem__(
+        self, extension_handle: _ExtensionFieldDescriptor[_ContainerMessageT, _ExtenderMessageT]
+    ) -> _ExtenderMessageT: ...
+    def __setitem__(
+        self, extension_handle: _ExtensionFieldDescriptor[_ContainerMessageT, _ExtenderMessageT], value: _ExtenderMessageT
+    ) -> None: ...
+    def __delitem__(self, extension_handle: _ExtensionFieldDescriptor[_ContainerMessageT, _ExtenderMessageT]) -> None: ...
+    def __contains__(self, extension_handle: _ExtensionFieldDescriptor[_ContainerMessageT, _ExtenderMessageT]) -> bool: ...
+    def __iter__(self) -> Iterator[_ExtensionFieldDescriptor[_ContainerMessageT, Any]]: ...
```

Comment:
Can this be `_ExtensionFieldDescriptor[_ContainerMessageT, _ExtenderMessageT]` here rather than `Any`?

If so, lets do it!
If not, can you leave a comment explaining it?

Reply:
I don't believe so.  With `__iter__` there wouldn't be enough information to resolve the `_ExtenderMessageT` type var.  With the other methods, there's an `_ExtensionFieldDescriptor` explicitly defined which serves this role.  Concretely, if we replace `Any` with `_ExtenderMessageT` then one of the tests I've written in `mypy-protobuf` fail with 
```
test/test_generated_mypy.py:356: error: Need type comment for 'x'  [var-annotated]
        for x in s1.Extensions:
        ^
```

Reply:
ah yeah - now I see.
It's a polymorphic dictionary, where the value type could be a variety of different types of _ExtensionFieldDescriptor - each entry would be for a different extension and have a different value type here.

`Any` sounds good to me!